version: "3.8"

services:
  3x-ui:
    image: ghcr.io/mhsanaei/3x-ui:latest
    environment:
      XRAY_VMESS_AEAD_FORCED: "false"
    volumes:
      - 3x-ui-db:/etc/x-ui/
      - 3x-ui-cert:/root/cert/
    networks:
      - server
    ports:
      - '30000-30050'
    restart: unless-stopped

  vpn-pihole:
    image: pihole/pihole:latest
    environment:
      TZ: 'Europe/Moscow'
      DNSSEC: 'true'
      DNSMASQ_LISTENING: 'all'
      FTLCONF_LOCAL_IPV4: '10.0.1.14'
    volumes:
      - vpn-pihole:/etc/pihole
      - vpn-pihole-dnsmasq:/etc/dnsmasq.d
    networks:
      - server
    dns:
      - 127.0.0.1
      - 8.8.8.8
    restart: unless-stopped

volumes:
  3x-ui-db:
  3x-ui-cert:
  vpn-pihole:
  vpn-pihole-dnsmasq:

networks:
  server:
    external: true
